### PR TITLE
feat: [ENG-1663] Improve the pubsub healthcheck to ping the brokers

### PR DESCRIPTION
**Pull request by phturb**
Simply add a ping call to ping the brokers and verify that one is healthy and that the connection can be made

### CODE REVIEW

**Comment by phturb** [comment]
Test flakyness fix has been included in these changes since the tests would not pass in the CI without them. Preferably this would be done in a different CR, but you don't need a version tag for a test fix

File: pubsubx/kgox/admin_client_impl_test.go [hunk 1]
```diff
@@ -30,87 +30,107 @@ func TestKadminClient_DeleteTopicsWithRetryTopics(t *testing.T) {
...
-			defer kadmCl.Close()
-			kadmCl.DeleteTopics(context.Background(), topics...)
-		})
 		require.NoError(t, err)
-		assert.Contains(t, createdTopics, topics[0])
-		assert.Contains(t, createdTopics, topics[1])
+		for _, top := range topics {
+			assert.Contains(t, createdTopics, top)
+		}
+		time.Sleep(time.Second)
```
Comment (pobeaulieu1) [inline pubsubx/kgox/admin_client_impl_test.go:63]: Can we avoid sleeping here?

Comment (pobeaulieu1) [reply pubsubx/kgox/admin_client_impl_test.go:63]: like polling for the expected condition with a timeout.

File: pubsubx/kgox/pubsub.go [hunk 2]
```diff
@@ -172,7 +172,7 @@ func (p *PubSub) AdminClient() (pubsubx.PubSubAdminClient, error) {
 		return nil, errorx.InternalErrorf("failed to create kafka client: %v", err)
 	}
 
-	admClient := NewPubSubAdminClient(kadm.NewClient(wc), p.conf, p.defaultCreateTopicConfigEntries)
+	admClient := NewPubSubAdminClient(wc, p.conf, p.defaultCreateTopicConfigEntries)
```
Comment (pobeaulieu1) [inline pubsubx/kgox/pubsub.go:175]: Not part of this PR but I noticed a typo in the function bellow. `getContexLogger` -> `getContextLogger`

**Review by pobeaulieu1 — APPROVED** [review]

**Review by pobeaulieu1 — APPROVED** [review]
Small comment to avoid using sleep in general to keep our CI fast and efficient but otherwise LGTM!
